src/models/Balances.js

```diff
@@ -38,32 +38,58 @@ export type ChainTotalBalancesPerAccount = {
 };
 
 export type CategoryAssetsBalances = {|
-  wallet?: AssetsBalances,
-  deposits?: AssetsBalances,
-  investments?: AssetsBalances,
-  liquidityPools?: AssetsBalances,
-  rewards?: AssetsBalances,
+  wallet?: WalletAssetsBalances,
+  deposits?: DepositAssetBalance[],
+  investments?: InvestmentAssetBalance[],
+  liquidityPools?: LiquidityPoolAssetBalance[],
+  rewards?: WalletAssetsBalances,
 |};
 
-export type AssetBalance = {
+export type ServiceAssetBalance = {|
+  key: string,
+  service: string,
+  title: string,
+  value: BigNumber,
+  iconUrl: ?string,
+  change?: BigNumber,
+|};
+
+export type LiquidityPoolAssetBalance = {|
+  ...ServiceAssetBalance,
+  share: ?BigNumber,
+|};
+
+export type InvestmentAssetBalance = {|
+  ...ServiceAssetBalance,
+|};
+
+export type DepositAssetBalance = {|
+  ...ServiceAssetBalance,
+  currentApy: ?BigNumber,
+|};
+
+export type WalletAssetBalance = {|
   balance: string,
   symbol: string,
```

Comment:
I would suggest migrating that to BigNumber on first possible occasion, but it will probably require changes to the call sites as well.

Reply:
Totally agree on idea to migrate/refactor this.